prow/plugins/lgtm/lgtm.go

```diff
@@ -300,12 +331,49 @@ func handlePullRequest(gc ghLabelClient, pe github.PullRequestEvent, log *logrus
 		return nil
 	}
 
-	// Don't bother checking if it has the label...it's a race, and we'll have
-	// to handle failure due to not being labeled anyway.
 	org := pe.PullRequest.Base.Repo.Owner.Login
 	repo := pe.PullRequest.Base.Repo.Name
 	number := pe.PullRequest.Number
 
+	// Check if we have LGTM label
+	hasLGTM := false
+	labels, err := gc.GetIssueLabels(org, repo, number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repo, number)
+	}
+	hasLGTM = github.HasLabel(lgtmLabel, labels)
+
+	if hasLGTM == true {
+		// Check if we have a tree-hash comment
+		var lastLgtmTreeHash string
+		botname, err := gc.BotName()
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get bot name.")
+		}
+		comments, err := gc.ListIssueComments(org, repo, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get the list of issue comments on %s/%s#%d.", org, repo, number)
```

Comment:
logger should get the metadata by default from context, you can omit the org repo number data here

Reply:
alright, I have just copied the log.WithError from line 312 here...
so, as a general rule, it's pointless to give this kind of details in logs?

Reply:
It's just redundant. When we construct the logger before passing the event off to a plugin, we seed the logger with a load of context. For instance, you can see with a message of `"Assigning $org/$repo#number to @author"` we get the data too from the logger context:

```
{"component":"hook","event-GUID":"97437ae0-ac60-11e8-8290-af2c9d8f78ff","event-type":"pull_request_review","level":"info","msg":"Assigning openshift/console#470 to spadgett","org":"openshift","plugin":"lgtm","pr":470,"repo":"console","review":151021137,"reviewer":"spadgett","time":"2018-08-30T14:25:49Z","url":"https://github.com/openshift/console/pull/470#pullrequestreview-151021137"}
```

(and usually it's a computer reading these and doing so on structured data is easier)